web/store/helpers/storeHelper.ts

```diff
@@ -0,0 +1,58 @@
+import { V1ObjectMeta } from "@kubernetes/client-node";
+
+// resourceObject is an interface of stored resource.
+interface resourceObject {
+  metadata?: V1ObjectMeta;
+}
+
+// ResourceState represents a type of each resource's state.
+type ResourceState<T extends resourceObject> = Array<T>;
+
+// createResourceState returns a new resource store's state.
+export function createResourceState<T extends resourceObject>(
```

Comment:
Almost resources will use this page's methods to handle Events. (Pod is an exception.)
This is because the Pod needs to consider to `unscheduled pod`.